File: frontend/src/types/activity.ts
export interface Activity {
  id: number;
  wing_id: number;
  title: string;
  description: string;
  activity_date: string;
  faculty_coordinator: string | null;
  report_url: string | null;
  report_cloudinary_id: string | null;
  created_at: string;
}

export interface ActivityCreate {
  wing_id: number;
  title: string;
  description: string;
  activity_date: string;
  faculty_coordinator?: string;
}

export interface ActivityUpdate {
  title?: string;
  description?: string;
  activity_date?: string;
  faculty_coordinator?: string;
}

File: backend/app/schemas/activity.py
from pydantic import BaseModel
from datetime import datetime, date
from typing import Optional


class ActivityBase(BaseModel):
    title: str
    description: str
    activity_date: date
    faculty_coordinator: Optional[str] = None


class ActivityCreate(ActivityBase):
    wing_id: int


class ActivityUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    activity_date: Optional[date] = None
    faculty_coordinator: Optional[str] = None


class ActivityResponse(ActivityBase):
    id: int
    wing_id: int
    report_url: Optional[str] = None
    report_cloudinary_id: Optional[str] = None
    created_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True
